program/llvm-toolchain-ppl.md

```diff
@@ -0,0 +1,252 @@
+
+## OpenHW members/participants committed to participate in this project
+
+All OpenHW Group members are invited to contributed expertise to this project. At present we are aware of the interest and commitment of:
+
+1. Thales
+2. Embecosm
+
+## Project Manager and technical lead
+
+- TBD
+
+This is a much lighter role than a traditional project manager, since the requirements are fully defined through the hardware design process.  The primary roles will be:
+ tracking and reporting status to OpenHW Software TG and Technical WG;
+- identifying and resolving issues related to the outcome.
+
+## Project documents
+
+The following project documents will be created:
+- Preliminary Project Proposal (this document)
+
+  - including an initial project plan
+  - including an initial risk register
+
+- Project Proposal, an updated version of this document
+
+  - separate detailed project plan
+  - separate risk register
+
+## Virtual customer
+
+It is proposed that the OpenHW Group Cores TWG acts as a "virtual customer" to exercise the compiler as it is developed.  In particular, the CVA6 project will exercise the multi-XLEN and core functionality of the LLVM compiler in the CVA6 verification flow.
+
+In addition, while the tool chain will have been thoroughly tested, it will benefit from the OpenHW Group Cores TWG being able to use it with their reference applications.
+
+## Summary of requirements
+
+The requirements for the CVA6 platform are captured in (link TBD).
+
+## Explanation of why OpenHW should do this project
+
+A processor is only useful with a robust, up-to-date, proven compiler tool chain.  CORE-V, being a strict RISC-V derivative, can use a standard RISC-V compiler tool chain, but in these circumstances will not be able to take advantage of any of the added features of the CORE-V processors.  This project will provide a baseline tool chain allowing these features to be used (albeit without optimization) and simulated from the source code level.
+
+Furthermore, the licensing scheme of LLVM (Apache license) makes it particularly appealing to industrial players.  Because it does not require the disclosure of compiler source code, chip manufacturers and system integrators can better protect their differentiating Intellectual Property assets.  This licensing scheme also creates additional opportunities for software companies which can provide high added-value services in the area of compiler tuning for hardware/system-centric companies.
+
+## Industry landscape
+
+The "original" compiler used for the development of RISC-V cores is GCC.  However, LLVM offers a licensing scheme that is more appealing to the processor industry as it does not require the disclosure of critical vendor-specific intellectual property (IP) when distributing the software.  LLVM-based compiler chains appear to be at the foundation of many market leader offerings and it is expected that this will also be the case for RISC-V / CORE-V solutions.
```

Comment:
I discussed this at length with Zbigniew, since this is a controversial area. How about this text:

> The original compiler used for the development of RISC-V cores is GCC licensed under the GNU General Public License, requiring vendors to make source code available.  LLVM offers a "permissive" licensing scheme which some customers find more attractive, since it does not require the source to be made available.  As a result LLVM-based compiler chains appear to be the basis (in whole of in part) of a number of proprietary compilers.  The downside is that there is less pressure on companies to contribute back to LLVM development.  This project will give OpenHW Group members a choice over which approach to adopt.